src/Linker/Linker.cpp

```diff
@@ -462,6 +459,32 @@ class LinkerImpl final : public Linker
         return true;
     }
 
+    bool BuildIWD(const std::string& projectName, const ZoneDefinition& zoneDefinition, SearchPaths& assetSearchPaths) const
+    {
+        const fs::path iwdFolderPath(m_args.GetOutputFolderPathForProject(projectName));
+        auto iwdFilePath(iwdFolderPath);
+        const auto iwdFileName = zoneDefinition.m_name + ".iwd";
+        iwdFilePath.append(iwdFileName);
+
+        fs::create_directories(iwdFolderPath);
+
+        const auto iwdWriter = IWDWriter::Create(iwdFilePath.string(), &assetSearchPaths);
+        for (const auto& assetEntry : zoneDefinition.m_assets)
+        {
+            iwdWriter->AddFile(assetEntry.m_asset_name);
```

Comment:
Does that mean it does try to write every asset into the IWD with the asset name interpreted as the full path?

That would mean that you cannot just simply use the same zone files as for the fastfile, right?
Because in a zone file for a fastfile you would write:
```
image,test_image
```
and it would refer to `images/test_image.iwi`.

In the context of the IWD here it would probably throw a "File not found"?

Reply:
Hmm that is true. Would that mean IWDs should have their own type of definition that is only paths to files? Because you can plausibly load any kind of asset from raw in games such as CoD4/T4. I suppose it could be handled by making it so each asset type that doesn't have the full path in its name has a folder: for example image asset to images folder this way you wouldn't need to specify the images folder in the file name.

Reply:
I mean while it's true that the older games support a lot of assets from IWDs, they should be consistent in what assets they put in which files, right? For example MW2 does only use IWDs for iwi images and streamed sounds.

I think some older titles also have weapon files in IWDs.

The way i'd do it is just have as much as possible in fastfiles and not in IWDs. At least for the first iteration.

Or what do you think?

Reply:
Maybe I should add: The idea i had in mind on how to handle these other project types in the future was not to write different zone files necessarly.

I added a thing called "assetlists" for IPaks previously for example.
In the future i thought it might be neat to be able to refer to an assetlist of the written fastfile to just "put everything from the fastfile to an IWD or IPak".

Like Linker would link a fastfile mod.ff and create an assetlists/mod.csv file which can then be refered to from the IWD zone file with `assetlist,mod`.

That would require the way you define a zone and the way you define an IWD to match though :sweat_smile: